controllers/managedocs_controller.go

```diff
@@ -1234,6 +1246,37 @@ func (r *ManagedOCSReconciler) reconcileOCSCSV() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileAdditionalPrometheusRule() error {
+	r.Log.Info("Reconciling Additional Prometheus Rule")
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.additionalRule, func() error {
+		if err := r.own(r.additionalRule); err != nil {
+			return err
+		}
+
+		desired := templates.AdditionalPrometheusRule.DeepCopy()
+
+		for _, group := range desired.Spec.Groups {
+			if group.Name == "pvc-uninstall-stuck-alert" {
+				for _, rule := range group.Rules {
+					if rule.Alert == "UninstallStuckDueToPVC" {
+						rule.Labels["namespace"] = r.namespace
```

Comment:
I see you are adding only the namespace here, any specific reason why we are filtering by group name and alertname?. Can we simply add namespace to all existing rules?

Reply:
We only have two Prometheus rules in the deployer. First one is DMS and second one is this additional prometheus rule. DMS gets labeled in reconcileDMSPrometheusRule and AdditionalPrometheusRule get labeled in reconcileAdditionalPrometheusRule.

Reply:
I agree that both rules are labeled in different methods. My question was why do we check for `group.Name` to be `pvc-uninstall-stuck-alert` and `rule.Alert` to be `UninstallStuckDueToPVC`. Any reason on why we are doing this?. Why can't we just traverse through all the groups and alerts in `AdditionalPrometheusRule` and label with namespace?.

Reply:
Yes, I can do that. Anyways all the additionalPrometheusRule will need the namespace label